gradle-plugin/src/test/kotlin/com/google/devtools/ksp/gradle/SourceSetConfigurationsTest.kt

```diff
@@ -69,6 +69,101 @@ class SourceSetConfigurationsTest {
         )
     }
 
+    @Test
+    fun registerJavaSourcesToAndroid() {
+        testRule.setupAppAsAndroidApp()
+        testRule.appModule.dependencies.addAll(
+            listOf(
+                module("ksp", testRule.processorModule),
+                module("kspTest", testRule.processorModule),
+                module("kspAndroidTest", testRule.processorModule)
+            )
+        )
+        testRule.appModule.buildFileAdditions.add(
+            """
+            tasks.register("printSources") {
+                fun logVariantSources(variants: DomainObjectSet<out com.android.build.gradle.api.BaseVariant>) {
+                    variants.all {
+                        println("VARIANT:" + this.name)
+                        val baseVariant = (this as com.android.build.gradle.internal.api.BaseVariantImpl)
+                        val variantData = baseVariant::class.java.getMethod("getVariantData").invoke(baseVariant)
+                            as com.android.build.gradle.internal.variant.BaseVariantData
+                        variantData.extraGeneratedSourceFolders.forEach {
```

Comment:
You can get all Java sources with `BaseVariant.getSourceFolders(SourceKind.JAVA)` and java compile classpath with ``BaseVariant.getCompileClasspath(null)`. It should be more stable across versions.

Reply:
when i was trying this, it was not included in `getSourceFolders` so i assumed it is only for regular sources, not generated sources.

Reply:
kk, this seems ok, and if the AGP API changes, it should be easy to migrate.